### ISSUE TITLE
option or behavior to ignore .pyc files that have no .py file

### ISSUE THREAD

**Issue by zzzeek** [ariebovenberg/slotscheck#57]
my git working repo leaves .pyc files around sometimes when I switch between branches.  slotscheck tries to import these .pyc files and reports them as import errors.    As slotscheck seeks to live in the space where "static anaylsis" tools do, even though at the moment it's not in fact static, can it at least ignore pyc files that have no .py, or can there be an option for this?

**Comment by ariebovenberg** [comment]
Makes sense. It's pretty common to have stale .pyc files floating around.

I'll do some investigation to decide whether to implement a one-off for pycache files, or whether implementing an `include/exclude-files` would be better.

**Comment by ariebovenberg** [comment]
Some investigation later: the current mechanism for module discovery (`pkgutil.walk_packages`) does not support filtering by file suffix. This module walking logic would have to be re-implemented, and then we might as well add full file-filtering support.

For those stumbling into this issue, a workaround is to clear all python cache files manually with something like `find . | grep -E "(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf`

**Comment by zzzeek** [comment]
yes that or use PYTHONDONTWRITEBYTECODE.   not an option I like very much but it's popular.

on my end, slotscheck is in our CI job so I'm not really going to be running it very often, there's no issue on this end.

I would think you could still use pkgutil, as you receive the output modules you could just look at module.__file__ to see if it's .pyc or .py (or if it is always .py, check os.path.exists(module.__file__)).
